src/main/java/com/uwetrottmann/tmdb2/TmdbInterceptor.java

```diff
@@ -39,10 +70,95 @@ public static Response handleIntercept(Chain chain, String apiKey) throws IOExce
         HttpUrl.Builder urlBuilder = request.url().newBuilder();
         urlBuilder.setEncodedQueryParameter(Tmdb.PARAM_API_KEY, apiKey);
 
+        if (request.url().pathSegments().get(0).equals("account")) {
```

Comment:
So in the TMDb API we have 2 authentication strategies. One strategy is with a guest_session, and one is with account_session, in any case, If a user tries to access Lists and Account endpoints will require an account_session, on the ratings though there are conflicts. Whether it should use a guest_session or account_session. You can't resolve these conflicts directly without a user's intervention. And while a user can use simultaneously both Guest Sessions and Account Sessions, it's a challenge to which session should be used for the rating, What i thought is either to create more methods declarations in the interface, provide an invalid url, masked one or either providing an invalid parameter, with an enum of the authentication strategy  , which the interceptor will determine what to do exactly. Is there any other idea on the table?